Corgibytes.Freshli.Cli.Test/Common/TestHelpers.cs

```diff
@@ -0,0 +1,17 @@
+using System.CommandLine;
+using System.Linq;
+using FluentAssertions;
+
+namespace Corgibytes.Freshli.Cli.Test.Common;
+
+public class TestHelpers
+{
+    public static void VerifyAlias<T> (string alias, ArgumentArity arity, bool allowMultipleArgumentsPerToken) where T: Command, new()
+    {
+        T command = new T();
+        Option option = command.Options.FirstOrDefault(x => x.Aliases.Contains(alias));
+        option.Should().NotBeNull();
+        option.AllowMultipleArgumentsPerToken.Should().Be(allowMultipleArgumentsPerToken);
+        option.Arity.Should().BeEquivalentTo(arity);
+    }
+}
```

Comment:
You didn't create this, so I think this little bit of clean-up can be pushed into a code clean-up issue.

Since we're using the `FluentAssertions` library, it would be cleanest to rely on [its support for adding new matchers](https://fluentassertions.com/extensibility/).

So this would end up looking like something like this: (I haven't tried to compile this.)

```csharp
public static class Command
{
    public static CommandOptionAssertions ShouldContainOptionAlias(this Command command, string alias)
    {
        return new CommandAssertions(command, alias);
    }
}

public class CommandOptionAssertions: ReferenceTypeAssertions<Option, CommandOptionAssertions>
{
    private string alias;
    private Command command;
    public CommandOptionAssertions(Command command, string alias): base(command)
    {
        this.command = command;
        this.alias = alias;
    }

    public AndConstraint<CommandAssertions> DoesAllowMultipleArgumentsPerToken()
    {
        Option option = command.Options.FirstOrDefault(x => x.Aliases.Contains(alias));
        Execute.Assertion
            .ForCondition(option != null)
            .FailWith("Alias with name {0} was not found", _ => alias)
            .Then
            .ForCondition(option.AllMultipleArgumentsPerToken)
            .FailWith("Expected to support multiple arguments per token")

        return new AndConstraint<CommandAssertions>(this);
    }

    public AndConstraint<CommandAssertions> DoesNotAllowMultipleArgumentsPerToken()
    {
        Option option = command.Options.FirstOrDefault(x => x.Aliases.Contains(alias));
        Execute.Assertion
            .ForCondition(option != null)
            .FailWith("Alias with name {0} was not found", _ => alias)
            .Then
            .ForCondition(!option.AllMultipleArgumentsPerToken)
            .FailWith("Expected to _not_ support multiple arguments per token")

        return new AndConstraint<CommandAssertions>(this);
    }

    public AndConstraint<CommandAssertions> HavingArity(int arity)
    {
        Option option = command.Options.FirstOrDefault(x => x.Aliases.Contains(alias));
        Execute.Assertion
            .ForCondition(option != null)
            .FailWith("Alias with name {0} was not found", _ => alias)
            .Then
            .BecauseOf(arity)
            .ForCondition(option.Arity == arity)
        return new AndConstraint<CommandAssertions>(this);
    }
}

```

And then I'm pretty sure it would be invoked like this:

```csharp
[Theory]
[InlineData("--output")]
[InlineData("-o")]
public void Verify_output_options_configuration(string alias)
{
    (new ScanCommand())
        .ShouldHaveOptionWithAlias("-o")
        .HavingArity(ArgumentArity.OneOrMore)
        .DoesAllowMultipleArgumentsPerToken();
}
```

Reply:
Let's definitely make a separate task for this and move the notes there. I'm wary of trying to implement this right now.

Reply:
Sounds good. Issue #68 has been created to track this.